vms/platformvm/utxo/camino_locked_test.go

```diff
@@ -657,3 +662,787 @@ func TestVerifyLockUTXOs(t *testing.T) {
+
+	testHandler := &handler{
+		ctx: ctx,
+		clk: &mockable.Clock{},
+		utxosReader: avax.NewUTXOState(
+			memdb.New(),
+			txs.Codec,
+		),
+		fx: fx,
+	}
+	txID := ids.GenerateTestID()
+	depositedAmount := uint64(2000)
+	outputOwners := defaultOwners()
+	depositedUTXOs := []*avax.UTXO{
+		generateTestUTXO(txID, ctx.AVAXAssetID, depositedAmount, outputOwners, testID, ids.Empty),
+	}
+
+	type args struct {
+		state        func(*gomock.Controller) state.Chain
+		keys         []*crypto.PrivateKeySECP256K1R
+		depositTxIDs []ids.ID
+	}
+	sigIndices := []uint32{0}
+
+	tests := map[string]struct {
+		args  args
+		want  []*avax.TransferableInput
+		want1 []*avax.TransferableOutput
+		want2 [][]*crypto.PrivateKeySECP256K1R
+		err   error
+	}{
+		"Error retrieving unlockable amounts": {
+			args: args{
+				state: func(ctrl *gomock.Controller) state.Chain {
+					nowMinus10m := uint64(time.Now().Add(-10 * time.Minute).Unix())
+					s := state.NewMockChain(ctrl)
+					deposit1 := deposit.Deposit{
+						DepositOfferID:      testID,
+						UnlockedAmount:      0,
+						ClaimedRewardAmount: 5000,
+						Start:               nowMinus10m,
+						Duration:            uint32((10 * time.Minute).Seconds()),
+					}
+					depositTxSet := ids.NewSet(1)
+					depositTxSet.Add(testID)
+
+					s.EXPECT().GetDeposit(testID).Return(&deposit1, nil)
+					s.EXPECT().GetDepositOffer(testID).Return(&deposit.Offer{
+						UnlockHalfPeriodDuration: uint32((5 * time.Minute).Seconds()),
+						Start:                    nowMinus10m,
+					}, nil)
+					s.EXPECT().LockedUTXOs(depositTxSet, gomock.Any(), locked.StateDeposited).Return(nil, fmt.Errorf("%w: %s", state.ErrMissingParentState, testID))
+					return s
+				},
+				keys:         preFundedKeys,
+				depositTxIDs: []ids.ID{testID},
+			},
+			err: fmt.Errorf("%w: %s", state.ErrMissingParentState, testID),
+		},
+		"Successful unlock of 50% deposited funds": {
+			args: args{
+				state: func(ctrl *gomock.Controller) state.Chain {
+					nowMinus10m := uint64(time.Now().Add(-10 * time.Minute).Unix())
+					s := state.NewMockChain(ctrl)
+					deposit1 := deposit.Deposit{
+						DepositOfferID:      testID,
+						UnlockedAmount:      0,
+						ClaimedRewardAmount: 5000,
+						Start:               nowMinus10m,
+						Duration:            uint32((10 * time.Minute).Seconds()),
+						Amount:              depositedAmount,
+					}
+					depositTxSet := ids.NewSet(1)
+					depositTxSet.Add(testID)
+
+					s.EXPECT().GetDeposit(testID).Return(&deposit1, nil)
+					s.EXPECT().GetDepositOffer(testID).Return(&deposit.Offer{
+						UnlockHalfPeriodDuration: uint32((5 * time.Minute).Seconds()),
+						Start:                    nowMinus10m,
+					}, nil)
+					s.EXPECT().LockedUTXOs(depositTxSet, gomock.Any(), locked.StateDeposited).Return(depositedUTXOs, nil)
+					return s
+				},
+				keys:         []*crypto.PrivateKeySECP256K1R{preFundedKeys[0]},
+				depositTxIDs: []ids.ID{testID},
+			},
+			want: []*avax.TransferableInput{{
```

Comment:
is it possible to use helper funcs here? like "generateInput" or something
they should be presented in helper file